lib/csf-tools/src/CsfFile.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable no-underscore-dangle */
+import fs from 'fs-extra';
+import { parse } from '@babel/parser';
+import generate from '@babel/generator';
+import * as t from '@babel/types';
+import traverse, { Node } from '@babel/traverse';
+import { toId, isExportStory } from '@storybook/csf';
+
+const logger = console;
+
+interface Meta {
+  title?: string;
+  includeStories?: string[] | RegExp;
+  excludeStories?: string[] | RegExp;
+}
+
+interface Story {
+  id: string;
+  name: string;
+  parameters: Record<string, any>;
+}
+
+const getMeta = (declaration: any): Meta => {
+  // { title: 'asdf', includeStories: /.../ (or []), excludeStories: ... }
+  const meta: Meta = {};
+  declaration.properties.forEach((p: Node) => {
+    if (t.isObjectProperty(p) && t.isIdentifier(p.key)) {
+      if (p.key.name === 'title') {
+        meta.title = parseTitle(p.value);
+      } else if (['includeStories', 'excludeStories'].includes(p.key.name)) {
+        // @ts-ignore
+        meta[p.key.name] = parseIncludeExclude(p.value);
+      }
+    }
+  });
+  return meta;
+};
+
+const parseTitle = (value: any) => {
+  if (value.type === 'StringLiteral') {
+    return value.value;
+  }
+  return undefined;
+};
+
+function parseIncludeExclude(prop: any) {
+  const { code } = generate(prop, {});
+  // eslint-disable-next-line no-eval
+  return eval(code);
+}
+
+export class CsfFile {
+  _ast: Node;
+
+  _meta?: Meta;
+
+  _stories: Record<string, Story> = {};
+
+  constructor(ast: Node) {
+    this._ast = ast;
+  }
+
+  parse() {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    traverse(this._ast, {
+      ExportDefaultDeclaration: {
+        enter({ node }) {
+          if (t.isObjectExpression(node.declaration)) {
+            // export default { ... };
+            self._meta = getMeta(node.declaration);
+          } else if (
+            // export default { ... } as Meta<...>
+            t.isTSAsExpression(node.declaration) &&
+            t.isObjectExpression(node.declaration.expression)
+          ) {
+            self._meta = getMeta(node.declaration.expression);
+          }
+        },
+      },
+      ExportNamedDeclaration: {
+        enter({ node }) {
+          if (t.isVariableDeclaration(node.declaration)) {
+            // export const X = ...;
+            node.declaration.declarations.forEach((decl) => {
+              if (
+                t.isVariableDeclarator(decl) &&
+                t.isIdentifier(decl.id) &&
+                isExportStory(decl.id.name, self._meta)
```

Comment:
Is this going to break down if the meta is exported *after* the story? Should we just collect all the exports and then do the filtering at the end?